x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/IlmHealthIndicatorService.java

```diff
@@ -409,17 +412,29 @@ public boolean test(Long now, IndexMetadata indexMetadata) {
      * @param maxTimeOn  Maximum time that an index should spend on this step.
      * @param maxRetries Maximum number of times that a step should be retried.
      */
-    public record StepRule(String step, TimeValue maxTimeOn, long maxRetries) implements RuleConfig {
-        static StepRule stepRule(String name, TimeValue maxTimeOn) {
-            return new StepRule(name, maxTimeOn, 100);
+    public record StepRule(String step, TimeValue maxTimeOn, Long maxRetries) implements RuleConfig {
```

Comment:
Shall we override the canonical constructor and add a validation that either `maxTimeOn` or `maxRetries` are different than `null` ? And otherwise, if both are `null`, throw an `IllegalArgumentException` ?